operator/src/test/java/org/keycloak/operator/testsuite/integration/KeycloakDeploymentTest.java

```diff
@@ -158,20 +158,13 @@ public void testConfigInCRTakesPrecedence() {
         var defaultKCDeploy = getTestKeycloakDeployment(true);
 
         var valueSecretHealthProp = new ValueOrSecret("health-enabled", "false");
-        var valueSecretProxyProp = new ValueOrSecret("proxy", "reencrypt");
```

Comment:
`KC_PROXY` env var is removed if `KC_PROXY_HEADERS` is present which is for all the tests. So we can't use it here but that should be fixe, it's enough to test the logic with just one option.